cscs-checks/apps/quantumespresso/quantumespresso_check.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class QuantumESPRESSOCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        self.valid_prog_environs = ['builtin']
+        self.executable = 'pw.x'
+        self.executable_opts = ['-in', 'ausurf.in']
+
+        self.sanity_patterns = sn.all([
+            sn.assert_found(r'convergence has been achieved', self.stdout),
+        ])
```

Comment:
You override `sanity_patterns` completely in both subclasses. So what's the point of having this one?

Reply:
Thanks for the comment @vkarak: I'm sorry, I did not intend to override this `sanity_patterns`, but to append the ones in the subclasses to this first one, which should be a prerequisite